app/src/main/java/com/example/paritosh/stormy/Presenter.java

```diff
@@ -0,0 +1,41 @@
+package com.example.paritosh.stormy;
+
+import android.content.Context;
+import android.widget.Toast;
+
+import com.example.paritosh.stormy.databinding.ActivityMainBinding;
+import com.example.paritosh.stormy.model.CurrentWeather;
+import com.example.paritosh.stormy.model.CurrentWeatherDataBindingModel;
+
+public class Presenter {
+
+
+    private WeatherDataProvider weatherDataProvider = new WeatherDataProvider();
+
+    public void updateWeatherDetails(final ActivityMainBinding binding, Context context) {
+        final MainActivity mainActivity = (MainActivity) context;
+        if (Utils.isNetworkAvailable(context)) {
+            weatherDataProvider.getCurrentWeather(new WeatherDataProvider.OnWeatherApiResponse() {
+                @Override
+                public void onSuccess(CurrentWeather weather) {
+                    CurrentWeatherDataBindingModel model = Utils.convertCurrentWeatherToDataBindingModel(weather);
+                    binding.setWeather(model);
```

Comment:
create a method in MainActivity to render the view-model like this,

render(CurrentWeatherDataBindingModel viewModel) {
   binding.setWeather(viewModel);
}

and call that render method here